src/pawelak-arek/Fsd.Pawelak.Cs/Fsd.Arek.Cs3/Domain/Product/Product.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Fsd.Arek.Cs3.Domain
+{
+    public class Product
+    {
+        public int ProductId { get; }
+
+        public string Name { get; set; }
+
+        public decimal Price { get; set; }
+
+        public Product(Warehouse warehouse, string name, decimal price)
+        {
+            Name = name;
+            Price = price;
+            ProductId = CreateProductID(warehouse, name, price);
+        }
+
+        public int CreateProductID(Warehouse warehouse, string name, decimal price)
+        {
+            int id = 0;
+            bool state;
+            try
+            {
+                state = warehouse.WarehouseState.Exists(x => x.Name == name) && warehouse.WarehouseState.Exists(x => x.Price == price) ? true : false;
```

Comment:
ja bym to zrobił odwrotnie - dlaczego produkt "grzebie" po stanie magazynu. To magazyn powinien nadawać produktowi Id i ukrywać swój stan.  A u Ciebie jest odwrotnie - produkt przegląda stan magazynu i sam sobie przypisuje ID